barretenberg/cpp/src/barretenberg/ultra_honk/merge_prover.cpp

```diff
@@ -25,7 +25,16 @@ MergeProver::MergeProver(const std::shared_ptr<ECCOpQueue>& op_queue,
 {
     // Merge the current subtable (for which a merge proof is being constructed) prior to
     // procedeing with proving.
-    op_queue->merge(settings);
+    if (settings == MergeSettings::APPEND) {
+        info("appending now");
+        size_t last_subtable_size = op_queue->get_unmerged_subtable_size();
+        op_queue->merge(settings, ECCOpQueue::OP_QUEUE_SIZE - last_subtable_size);
```

Comment:
I think we should fix this as a constant and enforce its correctness with a standalone test